File: php/app/mvc/models/Note.php
<?php
namespace mvc\models;

use mvc\core\Model;
use PDO;

class Note extends Model {
    public function get_all_api() {
        return $this->db->query("SELECT * FROM notes")->fetchAll(PDO::FETCH_ASSOC);
    }

    public function get_all() {
        return $this->db->query("SELECT * FROM notes");
    }


    public function get($note_id) {
        return $this->db->query("SELECT * FROM notes WHERE id = '$note_id';")->fetch(PDO::FETCH_BOTH);
    }

    public function get_api($note_id) {
        return $this->db->query("SELECT * FROM notes WHERE id = '$note_id';")->fetch(PDO::FETCH_ASSOC);
    }
    public function update($note_id, $title, $body) {
        $sql = "UPDATE `notes` SET `title` =  \"$title\", `body` = \"$body\" WHERE `id` = '$note_id'";
        $this->db->query($sql);
    }

    public function create($user_id, $title, $body) {
        $sql = "INSERT INTO `notes` (`user_id`, `title`, `body`) VALUES ($user_id, \"$title\", \"$body\")";
        $this->db->query($sql);
    }

    public function delete($note_id) {
        $this->db->query("DELETE FROM notes WHERE id = '$note_id';");
    }
}

File: server/mvc/repositories/note.py
import datetime

from mvc.models import Note
from mvc.models.base import db


class NoteRepository:

    def __init__(self):
        pass

    @staticmethod
    def get_all(note_user_id):
        return db.session.execute(db.select(Note).filter(Note.note_user_id == note_user_id)).scalars()

    @staticmethod
    def get(note_id, note_user_id):
        return db.session.execute(db.select(Note).filter(Note.note_user_id == note_user_id,
                                                         Note.id == note_id)).scalar()

    @staticmethod
    def create(note_user_id, title, description):
        note = Note(note_user_id=note_user_id, title=title, description=description)
        db.session.add(note)
        db.session.commit()
        return note

    @staticmethod
    def update(note_id, note_user_id, title, description):
        print(note_id, note_user_id)
        note = NoteRepository.get(note_id, note_user_id)
        print(note)
        if note:
            note.title = title
            note.description = description
            note.updated_timestamp = datetime.datetime.now()
            db.session.add(note)
            db.session.commit()
            return note

    @staticmethod
    def delete(note_id, note_user_id):
        note = NoteRepository.get(note_id, note_user_id)
        if note:
            db.session.delete(note)
            db.session.commit()
            return note
